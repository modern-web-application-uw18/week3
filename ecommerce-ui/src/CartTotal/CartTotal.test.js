import React from 'react';
import ReactDom from 'react-dom';
import TestRenderer from 'react-test-renderer';
import CartTotal from './CartTotal.js'

describe('CartTotal', () => {
    it('should clear smoke test for CartTotal rendering', () => {
        const div = document.createElement('div');
        const dummyProp = [
            {
                "title": "Cardboard Box",
                "houseType": "Apartment",
                "image": "./itsABox.jpg",
                "location": {
                    "city": "Seattle",
                    "country": "USA"
                },
                "payment": {
                    "cost": 1,
                    "description": "Free cancellation"
                },
                "host": {
                    "name": "I5 underpass",
                    "isSuperhost": false
                },
                "rating": {
                    "stars": 1,
                    "reviews": 6876
                }
            }
        ];
        ReactDom.render(<CartTotal itemsInCart={dummyProp}/>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('should render a minimal CartTotal section of Cart', () => {
        const dummyProp = [
            {
                "title": "Cardboard Box",
                "houseType": "Apartment",
                "image": "./itsABox.jpg",
                "location": {
                    "city": "Seattle",
                    "country": "USA"
                },
                "payment": {
                    "cost": 1,
                    "description": "Free cancellation"
                },
                "host": {
                    "name": "I5 underpass",
                    "isSuperhost": false
                },
                "rating": {
                    "stars": 1,
                    "reviews": 6876
                }
            }
        ];
        const component = TestRenderer.create(<CartTotal itemsInCart={dummyProp}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
})