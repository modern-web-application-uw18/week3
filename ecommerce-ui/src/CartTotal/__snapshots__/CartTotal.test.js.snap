// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotal should render a minimal CartTotal section of Cart 1`] = `
<strong>
  Total: $
  1
</strong>
`;
